test/graph/test_graph.cpp

```diff
@@ -0,0 +1,154 @@
+#include <vector>
+
+#include "graph/graph.h"
+#include "gtest/gtest.h"
+
```

Comment:
As for `ASSERT_NO_THROW`:

It seems you're generally not using exceptions. I think it is better to check some data about the graph in the assertion part. For example, if you have added one vertex, you can check that amount of vertices in the graph is 1.